app/vmauth/limited_revers_proxy.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+// LimitedReversProxy represents default revers proxy with configurable limit
+type LimitedReversProxy struct {
+	reversProxy *httputil.ReverseProxy
```

Comment:
It would be great implementing the needed reverse proxy logic by ourselves on top of `net/http.Client` instead of using `httputil.ReverseProxy`. This will result in more robust implementation, which will be easier to tune and extend.

Reply:
implemented logic on top of `net/http.Client` in the `lib/reverseproxy` folder